### PR TITLE
Ccg 228 integrate various dissonant parts of projects into one cohesive slice

### PR DESCRIPTION
**Pull request by Smoresies**
Was changes all made pre-meeting. Brings together the player into a meaningful scene on it's own and creates a sandbox level.

### CODE REVIEW

File: scripts/player_animation_controller.gd [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class_name PlayerAnimationController
```
Comment (SllimRevilo) [inline scripts/player_animation_controller.gd:1]: this class should be on an entity class.

File: scripts/player_animation_controller.gd [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+extends AnimationPlayer
+
+@export var animated_sprite: AnimatedSprite2D
+@export var player: CharacterBody2D
+
+func change_dir(dir: float):
```
Comment (SllimRevilo) [inline scripts/player_animation_controller.gd:7]: this should be connected to on_direction_changed in the movement component

File: scripts/entity/components/movement_components/base_movement_component.gd [hunk 3]
```diff
@@ -92,7 +92,7 @@ func on_jump_input_cancelled() -> void:
 ## Calls the movement signals using the given velocity and previous velocity
 func _emit_movement_signals() -> void:
 	if(velocity.x == 0 && _previous_horizontal_velocity != 0):
-		on_horizontal_movement_cancelled.emit(velocity.x)
+		on_horizontal_movement_cancelled.emit()
```
Comment (SllimRevilo) [inline scripts/entity/components/movement_components/base_movement_component.gd:95]: Why did this lose its param?

Comment (Smoresies) [reply scripts/entity/components/movement_components/base_movement_component.gd:95]: We talked about this in vc. It only ever emits when velocity is 0, so it doesn't need a param.

File: scripts/player_animation_controller.gd [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+func walking(dir: float):
+	change_dir(dir)
+	
+	if player.is_on_floor():
+		play(&"walking")
+
+func idle():
+	if player.is_on_floor():
+		play(&"idle")
```
Comment (SllimRevilo) [inline scripts/player_animation_controller.gd]: I dont think we shoudl be doing animations like this. I believe there is an animation tree that is a state machine

File: prefabs/player.tscn [hunk 5]
```diff
@@ -306,43 +369,93 @@ Vector2i(1, 1): {
...
+
+[node name="PlayerAttackComponent" type="Node" parent="." node_paths=PackedStringArray("lock_on_area", "attacks", "measurement_position", "projectile_spawn_position")]
+script = ExtResource("31_a0bar")
+lock_on_area = NodePath("../PlayerController/Area2D")
+attacks = [NodePath("../RangedAttack")]
+measurement_position = NodePath("../PlayerController/Marker2D")
+projectile_spawn_position = NodePath("../PlayerController/Marker2D")
+
+[connection signal="on_horizontal_movement" from="PlayerController" to="PlayerController/AnimatedSprite2D/AnimationPlayer" method="walking"]
+[connection signal="on_horizontal_movement_cancelled" from="PlayerController" to="PlayerController/AnimatedSprite2D/AnimationPlayer" method="idle"]
```
Comment (SllimRevilo) [inline prefabs/player.tscn:462]: I wouldnt hook up signals in group environment. It makes it hard to find connections when there is no way to reference them.

File: scripts/player_animation_controller.gd (hunk 4 above)
Comment (Smoresies) [reply scripts/player_animation_controller.gd]: Worth looking into for future projects, but is there any issue making the state machine in code for this one?

File: prefabs/player.tscn (hunk 5 above)
Comment (Smoresies) [reply prefabs/player.tscn:462]: What do you mean by this? This was from the player.tscn, this is just how I referenced the signals in the GUI

Comment (SllimRevilo) [reply prefabs/player.tscn:462]: The entity class was made specifically to deal with signals. 
In projects with more than one person it gets hard to mange signals in editor. I have done it in the past with Unity and if the person isnt available to tell you where it is then you cannot search it easily like you can in code

**Review by SllimRevilo — APPROVED** [review]

**Pull request merged**